config/configgrpc/configgrpc.go

```diff
@@ -287,6 +293,22 @@ func (gss *GRPCServerSettings) ToServerOption() ([]grpc.ServerOption, error) {
 		}
 	}
 
+	if gss.Auth != nil {
+		auth, err := auth.New(*gss.Auth)
+		if err != nil {
+			return nil, err
+		}
+
+		// perhaps we should use a timeout here?
+		if err := auth.Start(context.Background()); err != nil {
+			return nil, err
```

Comment:
This can go to configauth:

```
func (a *Auth) ToGrpcInterceptors() []grpc.DialOption
```

This way we don't expose internal things from configauth to configgrp

Reply:
For that, I had to move `internal/auth` to `configauth`, as it would create a non-allowed import cycle, as `internal/auth` depends on `configauth`, and `configauth` would start depending on `internal/auth`. This is something we discussed in another PR, and I think it makes sense, especially if we expand the configauth to accept external authenticators in the future.

This PR has been updated to reflect that.